packages/core/src/checkout-buttons/strategies/paypal-commerce/paypal-commerce-button-strategy.ts

```diff
@@ -60,6 +70,50 @@ export default class PaypalCommerceButtonStrategy implements CheckoutButtonStrat
         return Promise.resolve();
     }
 
+    private async _handleOnCancel(_data: OnCancelData) {
+        const lineItems = this._getLineItems();
+        const existingConsignments = this._store.getState().consignments.getConsignmentsOrThrow();
+        const { email } = this._store.getState().billingAddress.getBillingAddressOrThrow();
+        const { firstName, lastName, address1 } = existingConsignments?.[0].shippingAddress || {};
+        const shippingAddress = {
+            ...this._currentShippingAddress,
+            firstName: firstName !== 'Fake' ? firstName : '',
+            lastName: lastName !== 'Fake' ? lastName: '',
+            address1: address1 !== 'Fake street' ? address1 : '',
+            email: email !== 'fake@fake.fake' ? email : '',
+            address2: '',
+            stateOrProvince: '',
+            stateOrProvinceCode: '',
+            countryCode: this._currentShippingAddress?.countryCode || '',
+            postalCode: this._currentShippingAddress?.postalCode || '',
+            city: this._currentShippingAddress?.city || '',
+            phone: '',
+            company: '',
+            customFields: []
+        };
```

Comment:
As far as I see, we can make this code a bit simpler. 
`this._currentShippingAddress` - is a result of `_transformAddress` method. We are mocking almost all the data in `_transformAddress` method and we can skip mock duplication here.

In this case we can get next result:
 ```suggestion
            const shippingAddress = {
                ...this._currentShippingAddress,
                firstName: firstName !== 'Fake' ? firstName : '',
                lastName: lastName !== 'Fake' ? lastName: '',
                address1: address1 !== 'Fake street' ? address1 : '',
                email: email !== 'fake@fake.fake' ? email : '',
            };
```